### PR TITLE
SPJ-40: Add required fields for 3DSecure

### LINKED ISSUE #40: Add required fields for 3DSecure
purchaseRequest attribute in the formData is missing for 3DSecure

### PR DESCRIPTION
**Pull request by lcsfort**
- [x] Add tests cases
- [x] Update the documentation
- [x] Add a new function to handle purchases on Sisp class
- [x] Deprecate the old function to handle purchases

### CODE REVIEW

**Review by FabyoSK — DISMISSED** [review]

File: src/sisp.js [hunk 1]
```diff
@@ -151,17 +175,70 @@ class Sisp {
...
+
+      const error = validateUserBillingInfoSchema(userBillingInfo);
+
+      if (error) {
+        return error;
+      }
+
+      return this.#generateRequestForm(referenceId, total, webhookUrl, {
+        transactionCode: TRANSACTION_CODES.purchase,
+        is3DSec: '1',
```
Comment (Rony78K) [inline src/sisp.js]: put this magic number in a constant plz

**Review by Rony78K — DISMISSED** [review]

File: src/sisp.js (hunk 1 above)
Comment (lcsfort) [reply src/sisp.js]: Resolved

**Review by Rony78K — DISMISSED** [review]

**Review by FabyoSK — DISMISSED** [review]

**Review by vandro — DISMISSED** [review]

**Review by savylopes — DISMISSED** [review]

**Review by FredSoares — APPROVED** [review]

**Review by vandro — APPROVED** [review]

**Review by savylopes — APPROVED** [review]

**Review by FabyoSK — APPROVED** [review]

**Review by Rony78K — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -23,10 +23,99 @@ $ npm install @chuva.io/sisp
 $ yarn add @chuva.io/sisp
 ```
 ## Module Configuration
-Import `sisp-payments` and create a new instance using your credentials. (obtained from SISP):
+
+### Sisp with 3DSecure
+
+Import `@chuva/sisp/sisp3DS` and create a new instance using your credentials. (obtained from SISP):
```
Comment (trismegis7us) [inline README.md:29]: The import is incorrect

File: README.md [hunk 3]
```diff
@@ -23,10 +23,99 @@ $ npm install @chuva.io/sisp
...
+
+```js
+const Sisp = require('@chuva.io/sisp/sisp3DS');
+
+const posID = 900512;
```
Comment (trismegis7us) [inline README.md:34]: This should be a string.

File: README.md [hunk 4]
```diff
@@ -23,10 +23,99 @@ $ npm install @chuva.io/sisp
...
+const posAutCode = "123456789ssA";
+const url = "https://mc.vinti4net.cv/3ds_payments_url";
+
+const sisp = new Sisp({ posID, posAutCode, url });
+
+```
+#### Generate Payment Request Form with 3DSec
```
Comment (trismegis7us) [inline README.md:41]: Be consistent "3DSecure" as above.

File: src/sisp3DS/sisp.js [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+   * @param {String} userBillingInfo.workPhone.subscriber - Required - This indicates the phone number of the account holder, it should include the indicator without the + sign, example of a phone number of Cape Verd: 2389573234
+   * @param {Object} userBillingInfo.mobilePhone - Required - An Object containing information about personal phone number of the card holder
+   * @param {String} userBillingInfo.mobilePhone.cc - Required - This indicates the country code of the country of the phone number, example example - 123
+   * @param {String} userBillingInfo.mobilePhone.subscriber - Required - This indicates the phone number of the account holder, it should include the indicator without the + sign, example of a phone number of Cape Verd: 2389573234
+   * @param {String} options.languageMessages - Optional - This is used to setup the languages for the response messages.
+   * @param {Object} options.currencyCode - Optional - This is to setup the currency.
+   * @returns {Document} response - HTML Form to process payments
+   */
+  generatePaymentRequestForm = (referenceId, total, webhookUrl, userBillingInfo, options = { currencyCode: '', languageMessages: '' }) => {
```
Comment (trismegis7us) [inline src/sisp3DS/sisp.js:152]: Let's fix this interface to not allow empty strings as values.

Comment (trismegis7us) [inline src/sisp3DS/sisp.js:152]: See which parameters above can be better documented as enum types. I don't know what a valid currency code is, for example. `userBillingInfo.addrMatch` is another example.

**Review by trismegis7us — APPROVED** [review]

**Pull request merged**